pyk/src/pyk/proof/reachability.py

```diff
@@ -681,26 +679,12 @@ def _inject_module(module_name: str, import_name: str, sentences: list[KRuleLike
         _inject_module(proof.dependencies_module_name, self.main_module_name, dependencies_as_rules)
         _inject_module(proof.circularities_module_name, proof.dependencies_module_name, [circularity_rule])
 
-        self._check_all_terminals(proof)
+        if self.kcfg_explore.kcfg_semantics.is_terminal(proof.kcfg.node(proof.target).cterm):
+            proof.add_terminal(proof.target)
 
     def nonzero_depth(self, proof: APRProof, node: KCFG.Node) -> bool:
         return not proof.kcfg.zero_depth_between(proof.init, node.id)
 
-    def _check_terminal(self, proof: APRProof, node: KCFG.Node) -> None:
-        if node.id not in proof._checked_for_terminal:
-            _LOGGER.info(f'Checking terminal: {node.id}')
-            proof._checked_for_terminal.add(node.id)
-            if self.kcfg_explore.kcfg_semantics.is_terminal(node.cterm):
-                _LOGGER.info(f'Terminal node: {node.id}.')
-                proof.add_terminal(node.id)
-            elif self.fast_check_subsumption and self._may_subsume(proof, node):
-                _LOGGER.info(f'Marking node as terminal because of fast may subsume check {proof.id}: {node.id}')
-                proof.add_terminal(node.id)
```

Comment:
This was only used in `check_all_terminals`, and in other places we just use `is_terminal` of the underlying semantics, so I don't think we need it anymore.